staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -832,11 +832,7 @@ func (p *Parser) parseExactValue() (sets.String, error) {
 //  (5) A requirement with just !KEY requires that the KEY not exist.
 //
 func Parse(selector string) (Selector, error) {
-	parsedSelector, err := parse(selector)
-	if err == nil {
-		return parsedSelector, nil
-	}
-	return nil, err
+	return parse(selector)
```

Comment:
does this guarantee the returned Selector is nil when err is not nil?

Reply:
> does this guarantee the returned Selector is nil when err is not nil?

Function [parse](https://github.com/kubernetes/kubernetes/blob/92a14f4f7ef11c580961f0aebe491d0ff148e191/staging/src/k8s.io/apimachinery/pkg/labels/selector.go#L846) return `nil,err` when err is not ni,